### ISSUE TITLE
jQuery.get() always fails

### ISSUE THREAD

**Issue by alfonsomunozpomer** [tmpvar/jsdom#1248]
Hi, I was trying to test an AJAX request with jQuery and jsdom, but it doesn’t seem to work. Using `xmlhttprequest` works, though:

```
var jsdom = require("jsdom");
var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;

jsdom.env(
  '<html></html>',
  ["http://code.jquery.com/jquery.js"],
  function (err, window) {
    var $ = window.$;
    $.get("http://www.ebi.ac.uk/gxa/resources/html/help-tooltips.experiment-page.html", function(data) {
      console.log(data);
    }).fail(function() {
      console.log("Error");
    })
    .always(function() {
      console.log("Finished");
    });

    var xhr = new XMLHttpRequest();
    xhr.open('GET', 'http://www.ebi.ac.uk/gxa/resources/html/help-tooltips.experiment-page.html', true);

    xhr.onreadystatechange = function() {
        if(xhr.readyState == 4) {
            if(xhr.status == 200) {
                console.log(xhr.responseText);
            }
            else {
                console.log(xhr.status);
            }
        }
    };

    xhr.send();
  }
);
```

I always get `Error` with `$.get` no matter what. Is there anything I’m not doing right?

**Comment by Sebmaster** [comment]
Replace `.fail(function() {` with `.fail(function(e) {` and log `e` too so we actually know what's wrong.

**Comment by alfonsomunozpomer** [comment]
This is what I get with `console.log(e)`:
```
Object {
  readyState: 0,
  getResponseHeader: [Function],
  getAllResponseHeaders: [Function],
  setRequestHeader: [Function],
  overrideMimeType: [Function],
  statusCode: [Function],
  abort: [Function],
  state: [Function],
  always: [Function],
  then: [Function],
  promise: [Function],
  pipe: [Function],
  done: [Function],
  fail: [Function],
  progress: [Function],
  complete: [Function],
  success: [Function],
  error: [Function],
  status: 0,
  statusText: 'No Transport' }
```

**Comment by Sebmaster** [comment]
You might be able to fix that with `jQuery.support.cors = true;`.

**Comment by alfonsomunozpomer** [comment]
Thank you! It worked. :)

Is there any reason why `jQuery.support.cors` with jsdom is `false` by default?

**Comment by Sebmaster** [comment]
Apparently because of our incomplete XHR implementation, the check is [here](https://github.com/jquery/jquery/blob/99e8ff1baa7ae341e94bb89c3e84570c7c3ad9ea/src/ajax/xhr.js#L24).

**Comment by domenic** [comment]
Well, we should probably fix this...

**Comment by Sebmaster** [comment]
Heh, we should... fix a lot about our XHR implementation :confused:

**Comment by domenic** [comment]
This should be fixed by 2c8933905e9b81436843fe8d429918876a277e54 :)
